main/main.cpp

```diff
@@ -1287,8 +1293,8 @@ bool Main::start() {
 	for (int i = 0; i < args.size(); i++) {
 		//parameters that do not have an argument to the right
 		if (args[i] == "--no-docbase") {
-			doc_base = false;
 #ifdef TOOLS_ENABLED
```

Comment:
Please move the `--no-docbase` parsing inside `#ifdef TOOLS_ENABLED` too. Might require some tweaks to the `if`/`else if` chain for it work, maybe make this start with `if (args[i] == "--check-only")` outside `#ifdef TOOLS_ENABLED` (from line 1306).

Reply:
I mean something like this:
```diff
diff --git a/main/main.cpp b/main/main.cpp
index 5ccc6662e..9784005b4 100644
--- a/main/main.cpp
+++ b/main/main.cpp
@@ -1286,18 +1286,18 @@ bool Main::start() {
 	List<String> args = OS::get_singleton()->get_cmdline_args();
 	for (int i = 0; i < args.size(); i++) {
 		//parameters that do not have an argument to the right
-		if (args[i] == "--no-docbase") {
-			doc_base = false;
+		if (args[i] == "--check-only") {
+			check_only = true;
 #ifdef TOOLS_ENABLED
 		} else if (args[i] == "-e" || args[i] == "--editor") {
 			editor = true;
 		} else if (args[i] == "-p" || args[i] == "--project-manager") {
 			project_manager = true;
+		} else if (args[i] == "--no-docbase") {
+			doc_base = false;
 #endif
 		} else if (args[i].length() && args[i][0] != '-' && game_path == "") {
 			game_path = args[i];
-		} else if (args[i] == "--check-only") {
-			check_only = true;
 		}
 		//parameters that have an argument to the right
 		else if (i < (args.size() - 1)) {
```

Reply:
Thanks!